akka-http/src/main/scala/akka/http/javadsl/unmarshalling/sse/EventStreamUnmarshalling.scala

```diff
@@ -8,21 +8,30 @@ package unmarshalling
 package sse
 
 import akka.NotUsed
+import akka.actor.ActorSystem
 import akka.http.javadsl.model.HttpEntity
 import akka.http.javadsl.model.sse.ServerSentEvent
 import akka.stream.javadsl.Source
 
 /**
- * Using `fromEventStream` lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of
+ * Using `fromEventsStream` lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of
  * `ServerSentEvent`s.
  */
 object EventStreamUnmarshalling {
 
   /**
-   * Lets a `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
+   * Lets an `HttpEntity` with a `text/event-stream` media type be unmarshalled to a source of `ServerSentEvent`s.
    */
+  def fromEventsStream(implicit system: ActorSystem): Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]] =
+    scaladsl.unmarshalling.sse.EventStreamUnmarshalling.fromEventsStream(system)
+      .map(_.map(_.asInstanceOf[ServerSentEvent]).asJava)
+      .asInstanceOf[Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]]]
+
+  // for binary-compatibility, since 10.0.15
+  @deprecated("Binary compatibility method. Invocations should have an implicit ActorSystem in scope to provide access to configuration", "10.0.15")
   val fromEventStream: Unmarshaller[HttpEntity, Source[ServerSentEvent, NotUsed]] =
```

Comment:
Is it possible to remove the code duplication here by passing in default values somehow? Maybe add another overload for `fromEventsStream` here as well that takes `ServerSentEventSettings` as @raboof suggested in the other class.

Reply:
Couldn't see a way to deduplicate with default values without broadening the access of deprecated fields. So instead I created a private helper method for the mapping and casting.